src/bicgstab.cpp

```diff
@@ -108,7 +108,7 @@ static void xpay(int n, realnum *x, double a, const realnum *y) {
 typedef realnum *prealnum; // grr, ISO C++ forbids new (double*)[...]
 
 /* BiCGSTAB(L) algorithm for the n-by-n problem Ax = b */
-int bicgstabL(const int L, const int n, realnum *x,
+meep::integer bicgstabL(const meep::integer L, const meep::integer n, realnum *x,
```

Comment:
The return type is not just an error code (see below) but could also be the required workspace size which is derived from `n`. The type of  `n` has to be meep::integer because at the call site it is derived from various chunk specific sizes which were all promoted to meep::integer. `L` could stay an int. I probably just missed that.